pkg/knowledge_base2/properties/resource_property.go

```diff
@@ -0,0 +1,164 @@
+package properties
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/collectionutil"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+)
+
+type (
+	ResourceProperty struct {
+		DefaultValue *construct.ResourceId  `json:"default_value" yaml:"default_value"`
+		AllowedTypes construct.ResourceList `yaml:"allowed_types"`
+		Namespace    bool                   `yaml:"namespace"`
+		SharedPropertyFields
+		*knowledgebase.PropertyDetails
+	}
+)
+
+func (r *ResourceProperty) SetProperty(resource *construct.Resource, value any) error {
+	if val, ok := value.(construct.ResourceId); ok {
```

Comment:
nit: prefer a type switch. Also maybe support `*construct.Resource` even though we don't currently use it.

Reply:
yeah, im first going for parity, then we can beef these up and make them more robust